### PR TITLE
DBZ-7876 Add NATS user/password authentication config options

### PR DESCRIPTION
**Pull request by SylvainMarty**

### CODE REVIEW

File: debezium-server-nats-jetstream/src/main/java/io/debezium/server/nats/jetstream/NatsJetStreamChangeConsumer.java [hunk 1]
```diff
@@ -93,6 +101,8 @@ void connect() {
             if (jwt.isPresent() && seed.isPresent()) {
                 natsOptionsBuilder
                         .authHandler(Nats.staticCredentials(jwt.get().toCharArray(), seed.get().toCharArray()));
+            } else if (user.isPresent() && password.isPresent()) {
```
Comment (jpechane) [inline debezium-server-nats-jetstream/src/main/java/io/debezium/server/nats/jetstream/NatsJetStreamChangeConsumer.java:104]: @SylvainMarty Thanks fro the PR! Could you please run maven build locally as the code needs to be re-formatted?

Also one question - or more precisely doublechecking. Are we sure that the scenario where there is username but no password is not valid? I suppose so but I'd like if you can explicitly confirm i is the case.

Comment (SylvainMarty) [reply debezium-server-nats-jetstream/src/main/java/io/debezium/server/nats/jetstream/NatsJetStreamChangeConsumer.java:104]: @jpechane Of course, I'm going to run `maven build` locally and fix the formatting.

I found no evidence in the [official NATS documentation](https://docs.nats.io/using-nats/developer/connecting/userpass) that a username could be provided without a password. I think it is safe to assume this is not a supported way of authentication.
From a security point of view, only the password field supports encryption in NATS configuration so I don't see why NATS would officially supports a half-secured configuration.

Comment (SylvainMarty) [reply debezium-server-nats-jetstream/src/main/java/io/debezium/server/nats/jetstream/NatsJetStreamChangeConsumer.java]: @jpechane I pushed the code formatting made by Maven. 👍

Comment (SylvainMarty) [reply debezium-server-nats-jetstream/src/main/java/io/debezium/server/nats/jetstream/NatsJetStreamChangeConsumer.java]: CI is not passing for some reason and I don't think I can fix anything about this...
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  42:01 min
[INFO] Finished at: 2024-05-25T06:41:07Z
[INFO] ------------------------------------------------------------------------
Error:  Failed to execute goal org.apache.maven.plugins:maven-failsafe-plugin:3.1.2:verify (verify) on project debezium-server-redis: 
Error:  
Error:  Please refer to /home/runner/work/debezium-server/debezium-server/server/debezium-server-redis/target/failsafe-reports for the individual test results.
```

**Comment by jpechane** [comment]
@SylvainMarty Squashed and applied, thanks! The failure is known transitive issue on CI.

**Comment by jpechane** [comment]
@SylvainMarty Could you please provide a docs PR with new config options?

**Comment by SylvainMarty** [comment]
@jpechane Docs PR available here: https://github.com/debezium/debezium/pull/5580

**Pull request merged**